### ISSUE TITLE
Pictures not syncing for every contacts

### ISSUE THREAD

**Issue by 2edff206** [owncloud/contacts#1142]
### Steps to reproduce
1. On your android 5 phone, sync your contacts to your owncloud account.
(i am using an app called Easydavfurandroid to create an owncloud contact account on my phone)

### Expected behaviour
Every contacts should sync with all their details

### Actual behaviour
All my contacts are synced to my server but only a few of them have their pictures. For the rest of them, only their first letter are shown as contact picture.

### Server configuration

Raspbian Jessie and Apache (last version)

**Database:** Mysql

**PHP version:** last version

**ownCloud version:** 8.2.2

**Updated from an older ownCloud or fresh install:** Fresh install

Please note that I have applied this fix : https://github.com/gvde/contacts/commit/3dadadbfb9ce4d5b6b3437b49e06656f4bbb0caa and it didn't help

**List of activated apps:** contacts, notes, external storage, files, activity, gallery

**The content of config/config.php:**

`<?php
$CONFIG = array (
  'instanceid' => 'xxxxxx',
  'passwordsalt' => 'xxxxxxx',
  'secret' => 'xxxxxxxx',
  'trusted_domains' => 
  array (
    0 => '192.168.1.25',
    1 => 'x.x.x.x',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'http://192.168.1.25/owncloud',
  'dbtype' => 'mysql',
  'version' => '8.2.2.2',
  'appstoreenabled' => true,
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/var/www/owncloud/apps',
      'url' => '/apps',
      'writable' => true,
    ),
  ),
  'installed' => true,
  'loglevel' => '0',
  'forcessl' => true,
  'mail_from_address' => 'Notification owncloud',
  'mail_smtpmode' => 'php',
  'maintenance' => false,
  'theme' => '',
  'trashbin_retention_obligation' => 'auto',
  'appstore.experimental.enabled' => false,
  'appcodechecker' => false,
  'memcache.local' => '\OC\Memcache\APC',
);

Thanks a lot

**Comment by gvde** [comment]
You have to apply the other open pull requests as well. Your cards probably use 4.0 vcards with uri photos...

**Comment by 2edff206** [comment]
Could you please tell me which one are you referring to ? I don't want to install wrong things. Thanks

**Comment by gvde** [comment]
There are two open pull requests regarding photos. The older one may already be in your current version, the newest one has not been merged, yet...

**Comment by 2edff206** [comment]
Sorry to bother you but are you talking about this ? https://github.com/owncloud/contacts/pull/1112
I am new to github and I don't clearly understand what is what.

Thanks a lot for your help

**Comment by gvde** [comment]
Look at https://github.com/owncloud/contacts/pulls and you'll see all open pull requests. There are two related to photos...

**Comment by 2edff206** [comment]
I have applied https://github.com/owncloud/contacts/pull/1112 https://github.com/owncloud/contacts/pull/1066 and https://github.com/owncloud/contacts/pull/1067 and it still doesn't work. Am I doing something wrong?

**Comment by gvde** [comment]
Did you sync your cards again? Without the patch the cards in the owncloud database are broken and nothing will fix those. Clear the address book and sync working cards into owncloud...

Otherwise you will have to post those broken cards from the database to see what's going on...

**Comment by 2edff206** [comment]
I deleted them all and I synced them again. Can you tell me which data do you want? And maybe the mysql query to get it ? thanks

**Comment by gvde** [comment]
First do
`select id,fullname from oc_contacts_cards;`
and find the id of a card which is broken. Then do
`select carddata from oc_contacts_cards where id = <id>;`
to get the card in the database.

**Comment by 2edff206** [comment]
For a broken card the data is : 
`| BEGIN:VCARD
VERSION:3.0
UID:bb45de93-209c-4088-9002-7470e473167a
FN:(REMOVED FULL NAME)
N:(REMOVED LAST NAME);(REMOVED NAME);;;
TEL;TYPE=cell:(REMOVEC NUMBER)
PRODID:-//ownCloud//NONSGML Contacts 0.5.0.0//EN
REV:2016-02-18T10:58:53+00:00
END:VCARD
 |`

And for a valid one it is : 

`| BEGIN:VCARD
VERSION:3.0
UID:22559b4e-499e-47de-92ee-8ec913717f4d
FN:(REMOVED FULL NAME)
N:(REMOVE LAST NAME);(REMOVED NAME);;;
TEL;TYPE=cell:+(REMOVED NUMBER)
NOTE:<HTCData><Facebook>id\\:(REMOVED FB ID)/friendof\\:(REMOVED FB ID)</Faceboo
 k><Twitter>id\\:(REMOVED TWITTER ID)/friendof\\:(REMOVED TWITTER ID)</Twitter></HTCData>
PHOTO;ENCODING=b;TYPE=jpeg:/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAIBAQEBAQIBAQEC (REMOVED BASE64 DATA)

PRODID:-//ownCloud//NONSGML Contacts 0.5.0.0//EN
REV:2016-02-18T10:55:38+00:00
END:VCARD
|`

**Comment by gvde** [comment]
The first card doesn't have a photo. It's not broken. It's missing.

Are you sure the original has a photo? Can you export the original card with photo from your Android device?

**Comment by 2edff206** [comment]
Hello, I checked on my phone, every contact with missing picture is using the picture from Whatsapp. So I think that owncloud does not sync pictures if they are from Whatsapp.
Is it the expected behaviour or is it a bug ?

**Comment by gvde** [comment]
Noone can tell without seeing the card from your phone. Again: export the original card from your phone and post it. Without the original card it's impossible to tell what is originally synced into owncloud.

Also all logs would be extremely helpful from when you sync it into owncloud.

Does Andoird "link" social media accounts with contacts? If that's the case the original card may not even have any photo in the card, but the photo is only obtained from the link to whatsapp...

**Comment by 2edff206** [comment]
Yes I was looking at it right now and the behaviour of my phone is quite strange. When I export the card from it there is indeed no picture for theses contacts, so the issue is not about owncloud. Sorry to have wasted your time...

But it would be great if there was some way to sync the picture linked to the contact to owncloud if someone could implement that :)

**Comment by gvde** [comment]
You still don't post one of those original cards, thus it's hard to say anything. I doubt there is anything the contacts app in owncloud could implement. If it's really a "linked" account on your Android than your android must take the picture from the linked account and copy it into the card before syncing. Owncloud doesn't have access to your whatsapp account nor can it influence what it receives from your phone...

But again: without seeing an original card exported from your phone it's all just guesswork...

**Comment by 2edff206** [comment]
Here is the original card for a contact without picture in owncloud:

`BEGIN:VCARD
VERSION:2.1
N:LASTNAME;NAME;;;
FN:NUMBER
TEL;CELL;PREF:+NUMBER
END:VCARD`

There is no picture. But on the contact app of my phone I can see a picture for this contact because I synced the pictures from Facebook with an app called Sync.me. Do you know any app that can copy the linked picture to the card?

**Comment by gvde** [comment]
Sorry but I don't know Android...

**Comment by 2edff206** [comment]
Thanks again for your help. Have a great night
